packages/core/src/api/resolvers/entity/product-variant-entity.resolver.ts

```diff
@@ -80,8 +80,13 @@ export class ProductVariantEntityResolver {
         @Parent() productVariant: ProductVariant,
     ): Promise<Product | undefined> {
         if (productVariant.product) {
-            return productVariant.product;
+            return this.requestContextCache.get(
+                ctx,
+                `ProductVariantEntityResolver.product(${productVariant.productId})`,
+                () => this.productVariantService.translateProduct(ctx, productVariant.product),
```

Comment:
I don't think we need to create a brand new method `.translateProduct()` for this purpose. We can reuse `getProductForVariant()` which is already doing the translation step.

Which then would make both branches do the same thing, which also doesn't make sense. So there must be a logical branch in which this additional lookup/translation does not need to be done - that would be if the `product` is already translated. This can be tested by inspecting `productVariant.product.name` to see if it is defined. If so, we should be ok to just return it as-is like in the original version.

Reply:
You're right, both branches are doing almost the same thing. I just pushed some changes. Also added the test.

To see the bug:

1. Check out commit [ac26c4a](https://github.com/vendure-ecommerce/vendure/pull/2644/commits/ac26c4a54b85a211d1efe950f72005b1f1d98802) 
2. `yarn build`
3. `yarn e2e`
4. test fails

Fix:

1. Checkout commit [368a9fc](https://github.com/vendure-ecommerce/vendure/pull/2644/commits/368a9fc1d325453fb99f5ef4b121b0a1046f38ce)
2. `yarn build`
3. `yarn e2e`
4. test does not fail